compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -1461,6 +1461,15 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                 }
 
                 TraitCandidate(trait_ref) => {
+                    if let Some(method_name) = self.method_name {
+                        // Some trait methods are excluded for arrays before 2021.
```

Comment:
Can we give an example here rather than 'some'? I think it is only for IntoIterator right?

Reply:
I was leaving the possibility that we might find more uses for this, but I've added a mention of `array.into_iter()` specifically.